src/Util.hs

```diff
@@ -20,10 +23,18 @@ extractLogsFromZip numberOfFiles file = do
     mTake n = Map.fromDistinctAscList . take n . Map.toAscList
 
 -- | Read zipe file
-readZip :: LByteString -> Either Text (Map FilePath LByteString)
+-- toArchiveOrFail is a partial function, we need to use tryDeep to catch the exception and throw it 
+-- upwards
+-- Why tryDeep instead of try? It's because we need to fully evaluate bytestrings in order to catch
+-- the decompression issue.
+readZip :: (MonadCatch m, MonadIO m) => LByteString -> m (Map FilePath LByteString)
 readZip rawzip = case Zip.toArchiveOrFail rawzip of
```

Comment:
Did you try it out? Does it catch the exceptions?

Reply:
Yes, I did try to use `try` at first since it has less constraints but it didn't catch the decompression error.
I can try again.